common/src/main/java/com/wynntils/functions/RaidFunctions.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.functions;
+
+import com.wynntils.core.components.Models;
+import com.wynntils.core.consumers.functions.Function;
+import com.wynntils.core.consumers.functions.arguments.FunctionArguments;
+import com.wynntils.models.raid.type.Raid;
+import com.wynntils.models.raid.type.RaidRoomType;
+import java.util.List;
+
+public class RaidFunctions {
+    public static class CurrentRaidFunction extends Function<String> {
+        @Override
+        public String getValue(FunctionArguments arguments) {
+            Raid currentRaid = Models.Raid.getCurrentRaid();
+
+            return currentRaid == null ? "" : currentRaid.getName();
+        }
+
+        @Override
+        protected List<String> getAliases() {
+            return List.of("raid");
+        }
+    }
+
+    public static class CurrentRaidRoomFunction extends Function<String> {
+        @Override
+        public String getValue(FunctionArguments arguments) {
+            if (Models.Raid.getCurrentRaid() == null) return "";
+            // Room should never be null if the raid is not but just in case the tracking fails
+            if (Models.Raid.getCurrentRoom() == null) return "";
+
+            return Models.Raid.getCurrentRoom().name();
+        }
+
+        @Override
+        protected List<String> getAliases() {
+            return List.of("raid_room");
+        }
+    }
+
+    public static class CurrentRaidTimeFunction extends Function<Integer> {
+        @Override
+        public Integer getValue(FunctionArguments arguments) {
+            if (Models.Raid.getCurrentRaid() == null) return -1;
+
+            return Models.Raid.currentRaidTime();
+        }
+
+        @Override
+        protected List<String> getAliases() {
+            return List.of("raid_time");
+        }
+    }
+
+    public static class CurrentRaidRoomTimeFunction extends Function<Integer> {
+        @Override
+        public Integer getValue(FunctionArguments arguments) {
+            if (Models.Raid.getCurrentRaid() == null) return -1;
+            // Room should never be null if the raid is not but just in case the tracking fails
+            if (Models.Raid.getCurrentRoom() == null) return -1;
+
+            return Models.Raid.currentRoomTime();
+        }
+    }
+
+    public static class RaidRoomTimeFunction extends Function<Integer> {
+        @Override
+        public Integer getValue(FunctionArguments arguments) {
+            if (Models.Raid.getCurrentRaid() == null) return -1;
+
+            RaidRoomType roomType =
+                    RaidRoomType.fromName(arguments.getArgument("roomName").getStringValue());
```

Comment:
It's a bit unfortunate that we can't provide a list of possible values for functions at the moment. Although it wouldn't be a big change, we would have to go through many functions to fill in the possible values. Aaand if we do that, we might as well add support for "enum-functions". So most likely out of scope for now.